torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1022,12 +1050,15 @@ void BuildLoweringContextSubmodule(py::module* m) {
 
   lowering_context_class.def(py::init<>())
       .def("build", &PyLoweringContext::Build)
+      .def("build_for_while", &PyLoweringContext::BuildForWhile)
       .def("hlo", &PyLoweringContext::GetHlo)
       .def("hlo_text", &PyLoweringContext::GetHloText)
       .def("hlo_json", &PyLoweringContext::GetHloJsonText)
       .def("parameter_id_tensor_mapping",
            &PyLoweringContext::GetParameterIdTensorMapping)
-      .def("tensor_parameter_id", &PyLoweringContext::GetTensorParameterId);
+      .def("tensor_parameter_id", &PyLoweringContext::GetTensorParameterId)
+      .def("setnamestring", &PyLoweringContext::SetNameString)
```

Comment:
Do you need this setter/getter for the name? Isn't there always one context/graph in your impl that you can name whatever? Also, "settnamestring" -> "set_name_string"

Reply:
Thanks, updated the function name in newer commit;

yes, we need this setter/getter for the name to recognize body_computation/cond_computation by name given when we create the lowering context; With recognized the body_computation/cond_computation by name, we could skip tuple for cond/body computation creation to match the xla::While [format requirement for cond](https://github.com/openxla/xla/blob/0f46dddc43c1b4989db8806a9b9abd128234e48a/xla/service/shape_inference.cc#L3167), and [error log](https://gist.github.com/ManfeiBai/19b6fc987a15a020a665e5498a446e8b)

and we have tried another solution like below, which modification could not be caught when create computation:
```
body_ctx_builder = body_ctx.builder()
body_ctx_builder.name_ = 'bodyctx'
```

so we choose to add this setter/getter function for xla::While requirement, and please feel free to redirect me to other possible solutions if I missed